redis/src/aio/pubsub.rs

```diff
@@ -297,28 +297,44 @@ impl PubSubSink {
     pub async fn subscribe(&mut self, channel_name: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("SUBSCRIBE");
         cmd.arg(channel_name);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Unsubscribes from channel.
     pub async fn unsubscribe(&mut self, channel_name: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("UNSUBSCRIBE");
         cmd.arg(channel_name);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Subscribes to a new channel with pattern.
     pub async fn psubscribe(&mut self, channel_pattern: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("PSUBSCRIBE");
         cmd.arg(channel_pattern);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Unsubscribes from channel pattern.
     pub async fn punsubscribe(&mut self, channel_pattern: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("PUNSUBSCRIBE");
         cmd.arg(channel_pattern);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
+    }
+
+    /// Sends a ping with a message to the server
```

Comment:
Can't you use builder style here and do `ping().message()` or just accept optional message? (I prefer the first)
Common in rust, and align with the real command name. Both are `ping` command.
Moreover, not a strong reason, but if tomorrow there's another optional parameter, you just add another option to the builder.

Reply:
there isn't a builder pattern for commands ATM, and given how long the ping command was unchanged, I'll be very susprised if this will be changed any time soon.
If we had ` ping()` object, what would it return? how would the object be executed?